---
title: C28125
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C28125
helpviewer_keywords:
- C28125
ms.assetid: 3f7b8db4-6a64-4480-919e-3f3ceca7f948
author: corob-msft
ms.author: corob
manager: markl
ms.workload:
- multiple
ms.openlocfilehash: 3b4d55d905208b1b05610aaae91806dc9a44562e
ms.sourcegitcommit: 68f893f6e472df46f323db34a13a7034dccad25a
ms.translationtype: MT
ms.contentlocale: zh-CN
ms.lasthandoff: 02/15/2020
ms.locfileid: "77264263"
---
# <a name="c28125"></a>C28125
警告 C28125：必须从 try/except 块中调用函数

 驱动程序正在调用一个函数，该函数必须从 try/except 块中调用，如[ProbeForRead](/windows-hardware/drivers/ddi/content/wdm/nf-wdm-probeforread)、 [ProbeForWrite](/windows-hardware/drivers/ddi/content/wdm/nf-wdm-probeforwrite)、 [MmProbeAndLockPages](/windows-hardware/drivers/ddi/content/wdm/nf-wdm-mmprobeandlockpages)。

## <a name="example"></a>示例
 下面的代码示例将生成此警告：

```cpp
ProbeForRead(addr, len, 4);
```

 下面的代码示例可避免此警告：

```cpp
__try
{
    ProbeForRead(addr, len, 4);
}
__except(EXCEPTION_EXECUTE_HANDLER)
{
    Status = GetExceptionCode();
    ... report error status
}
```
